spec/controllers/reports_controller_spec.rb

```diff
@@ -83,5 +94,46 @@
 
       it { is_expected.to render_template :letters }
     end
+
+    describe 'GET #raw_data' do
+      before { get :raw_data }
+
+      subject { response }
+
+      it { is_expected.to have_http_status(:success) }
+
+      it { is_expected.to render_template :raw_data }
+
+      describe 'assigns form object' do
+        subject { assigns(:form) }
+
+        it { is_expected.to be_a_kind_of Forms::FinanceReport }
+      end
+    end
+
+    describe 'PUT #raw_data' do
+      subject { response }
+
+      context 'with invalid data' do
+        before { put :raw_data_export, forms_finance_report: { date_from: nil, date_to: '2015-12-31' } }
+
+        it { is_expected.to have_http_status(:success) }
```

Comment:
As I understand it, yes, because it is re-rendering the input view with validation warnings on the affected fields